rest/config_test.go

```diff
@@ -1656,7 +1657,16 @@ func TestLoadJavaScript(t *testing.T) {
 			}()
 			js, err := loadJavaScript(inputJavaScriptOrPath, test.insecureSkipVerify)
 			if test.errExpected != nil {
-				require.True(t, errors.As(err, &test.errExpected)) // nolint: govet // CBG-2242
+				errx509Type := x509.UnknownAuthorityError{}
+				if test.errExpected == errx509Type {
+					expectedErrorString := test.errExpected.Error()
+					if runtime.GOOS == "darwin" {
+						expectedErrorString = "certificate is not trusted"
+					}
+					require.ErrorContains(t, err, expectedErrorString)
```

Comment:
Can we put this into a test helper function so it's shared for all the tests? e.g. `assertX509UnknownAuthority(err)`

Still _some_ amount of duplication to do the type switching on the expected error, but that's ok.